bionic/cache_api.py

```diff
@@ -0,0 +1,108 @@
+"""
+Contains the Cache class, which provides a user-facing API for directly interacting
+with Bionic's cache.
+"""
+
+from functools import total_ordering
+
+from .descriptors.parsing import entity_dnode_from_descriptor
+from .persistence import path_from_url, is_file_url
+
+
+class Cache:
+    """
+    A programmatic interface to Bionic's persistent cache.
+
+    Accessible as an attribute named ``cache`` on a :class:`Flow <bionic.Flow>`
+    object. Use ``get_entries`` to iterate through the set of cache entries.
+    """
+
+    def __init__(self, flow):
+        self._flow = flow
+
+    def get_entries(self):
+        """
+        Returns a sequence of :class:`CacheEntry <bionic.cache_api.CacheEntry>`
+        objects, one for each artifact in Bionic's persistent cache.
+        """
+
+        # These private accesses are a bit gross, but I'm not sure it's worth adding
+        # more layers of APIs to avoid them.
+        deriver = self._flow._deriver
+        deriver.get_ready()
+        persistent_cache = deriver._bootstrap.persistent_cache
+
+        stores = [
+            store
+            for store in [persistent_cache._local_store, persistent_cache._cloud_store]
+            if store is not None
+        ]
+
+        return (
+            CacheEntry(cache=self, inv_item=inv_item,)
+            for store in stores
+            for inv_item in store.inventory.list_items()
+        )
+
+
+def path_from_url_if_file(url):
```

Comment:
Not in this PR, but we should move all file related helpers together in a separate util.

Reply:
Agreed, I was thinking the same thing.